### ISSUE TITLE
RN 0.63.2 Fast Refresh is always on

### ISSUE THREAD

**Issue by gazedash** [facebook/react-native#29912]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description
Issue 1. Fast Refresh is always working, despite being turned off
Turning it "on" and "off" again doesn't change anything

Issue 2. Fast Refresh eats up a lot of resources on MacOS, especially if used for long periods of time
This can be confirmed if you kill the app, or disable it, stability of the system returns to normal

## React Native version:
Run `react-native info` in your terminal and copy the results here.
```
System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 37.62 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.3.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.7, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2, 29.0.3
      System Images: android-28 | Intel x86 Atom_64, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 11.7/11E801a - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_252 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.2 => 0.63.2 
```
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.
1. react-native init MyApp
2. react-native run-ios

## Expected Results
Describe what you expected to happen.
1. It is possible to enable and disable fast refresh, and the menu reflects it state
2. The fast refresh doesn't eat all the resources of the system that is running on

**Comment by fnando** [comment]
I have the same issue.

**Comment by laurent22** [comment]
Same here, impossible to disable Fast Refresh. I've also tried to disable it by code as described [here](https://stackoverflow.com/q/60729710/561309) but that didn't help. When a feature is that broken it shouldn't be enabled by default, or it should be very easy to switch it off.

**Comment by Barney4242** [comment]
Same here RN 0.63.4

**Comment by anil-sky-uk** [comment]
Hacky workaround to disable it completely for now...

In file: `node_modules/react-native/Libraries/Utilities/HMRClient.js`
return false on line 61

```
const HMRClient: HMRClientNativeInterface = {
  enable() {
    return false; // Line 61 - disable fast refresh : https://github.com/facebook/react-native/issues/29912
    // ...
```

**Comment by danlugo92** [comment]
I have the same issue.

```
System:
    OS: macOS 11.1
    CPU: (4) x64 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
    Memory: 307.57 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.18.2 - /var/folders/wn/lfk8lrp54lq7p4c2syp6_gnc0000gn/T/yarn--1611856822446-0.7514681844463749/node
    Yarn: 1.22.10 - /var/folders/wn/lfk8lrp54lq7p4c2syp6_gnc0000gn/T/yarn--1611856822446-0.7514681844463749/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v12.18.2/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.1, 30.0.2
      System Images: android-28 | Google Play Intel x86 Atom, android-29 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 12.3/12C33 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_265 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: ~0.63.4 => 0.63.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
✨  Done in 11.55s.

```

**Comment by soroush-b** [comment]
I noticed this is already fixed in 0.64.0 version and here is the PR https://github.com/facebook/react-native/commit/845e9eaafb08b4ca87a9987e840798e0ba011676 . 
you can either upgrade or add a patch to fix this. this worked for me.
